.github/workflows/check.yaml

```diff
@@ -0,0 +1,51 @@
+name: Check
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      ## Try getting the node modules from cache, if failed npm ci
+      - uses: actions/cache@v2
+        id: cache-npm
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.OS }}-node-${{ env.cache-name }}-
+            ${{ runner.OS }}-node-
+            ${{ runner.OS }}-
+      - name: Install npm deps
+        if: steps.cache-npm.outputs.cache-hit != 'true'
+        run: npm ci
+
+      - name: Commit linting
+        uses: wagoid/commitlint-github-action@v2
+
+      - name: Code linting
+        run: npm run lint:check
+        env:
+          CI: true
+
+      - name: Build nodejs
+        run: npm run compile -- --env mode=production
```

Comment:
the `npm run compile` calls two other commands, the parameter passing in this case may be without effect. I've tested with `npm run compile -- --env mode=development` did not work as excepted - the compilation happened in production mode. If this is the case just remove the `-- --env mode=production` part as production mode is the default.

Reply:
```suggestion
        run: npm run compile
```